src/containers/ModelPanel/index.tsx

```diff
@@ -81,10 +89,25 @@ const ModelPanel: React.FC<ModelPanelProps> = ({
     isNetworkedFile,
     changeToNetworkedFile: loadNetworkFile,
     recentColors,
-    applyUserColor,
     setRecentColors,
     selectedAgentMetadata,
+    setSelectedUIDisplayData,
+    setCurrentColorSettings,
+    storeDisplayDataInBrowser,
+    currentUIDisplayData,
 }): JSX.Element => {
+    const updateSelectedUiDisplayData = (colorChange: ColorChange) => {
+        const newUIData = applyColorChangeToUiDisplayData(
+            colorChange,
+            currentUIDisplayData
+        );
+        setSelectedUIDisplayData(newUIData);
+        setCurrentColorSettings({
+            currentColorSettings: ColorSettings.UserSelected,
+        });
+        storeDisplayDataInBrowser(newUIData);
+    };
+
```

Comment:
Calling all relevant redux action creators here in the function, and passing it down to `ColorPicker` as a callback. Also doing the conversion from `ColorChange` to `UIDisplayData` before hitting the `redux` plumbing to address some concerns raised by @frasercl in a previous PR about abuse of `redux-logic`.

I think there are other places in the repo where we drill action creators down as props (and sometimes also rename them several times) that are a bit confusing, maybe worth refactoring some of them, I like this approach of defining a callback as it keeps the `containers` as the only components that directly handle the redux functions.